clean-ci-resources.sh

```diff
@@ -62,22 +62,42 @@ openstack container list -f value -c Name \
 	| xargs --verbose --no-run-if-empty openstack container delete -r
 
 for resource in 'volume snapshot' 'volume' 'floating ip' 'security group' 'keypair' 'loadbalancer'; do
-	if [[ ${resource} == 'volume' ]]; then
-	  for r in $(./stale.sh -q "$resource"); do
-	    status=$(openstack "${resource}" show -c status -f value "${r}")
-		case "$status" in
-			# For Cinder volumes, deletable states are documented here:
-			# https://docs.openstack.org/api-ref/block-storage/v3/index.html?expanded=delete-a-volume-detail#delete-a-volume
-			available|in-use|error|error_restoring|error_extending|error_managing)
-				break
-				;;
-			*)
-				echo "${resource} ${r} in wrong state: ${status}, will try to set it to 'error'"
-				openstack "$resource" set --state error "$r" || >&2 echo "Failed to set ${resource} ${r} state to error, ${r} will probably fail to be removed..."
-				;;
-		esac
-	  done
-	fi
+	case $resource in
+		volume)
+		  for r in $(./stale.sh -q "$resource"); do
+			status=$(openstack "${resource}" show -c status -f value "${r}")
+			case "$status" in
+				# For Cinder volumes, deletable states are documented here:
+				# https://docs.openstack.org/api-ref/block-storage/v3/index.html?expanded=delete-a-volume-detail#delete-a-volume
+				available|in-use|error|error_restoring|error_extending|error_managing)
+					break
+					;;
+				*)
+					echo "${resource} ${r} in wrong state: ${status}, will try to set it to 'error'"
+					openstack "$resource" set --state error "$r" || >&2 echo "Failed to set ${resource} ${r} state to error, ${r} will probably fail to be removed..."
+					;;
+			esac
+		  done
+		  # shellcheck disable=SC2086
+          ./stale.sh -q $resource | report $resource | xargs --verbose --no-run-if-empty openstack $resource delete
+		  ;;
+		loadbalancer)
+		  for r in $(./stale.sh -q "$resource"); do
+			status=$(openstack "${resource}" show -c provisioning_status -f value "${r}")
+			case "$status" in
+				ERROR)
+					report $resource | xargs --verbose --no-run-if-empty openstack $resource delete
+					;;
+				*)
+					;;
+			esac
```

Comment:
one nice feature, that could be added later, is to report PENDING_* loadbalancers, that are staled. Not trying to remove them (technically not possible), but reporting them in the logs, so CI cop can take actions (manual cleanup).